dynamic/informer/informer.go

```diff
@@ -105,10 +107,20 @@ type sharedResourceInformer struct {
 	close func()
 }
 
+type dynamicResourceClientListFunc func(metav1.ListOptions) (*unstructured.UnstructuredList, error)
+
+func castToCacheListFunc(fn dynamicResourceClientListFunc) cache.ListFunc {
+	options := metav1.ListOptions{}
+	unstructuredList, err := fn(options)
+	return func(metav1.ListOptions) (runtime.Object, error) {
+		return runtime.Object(unstructuredList), err
+	}
+}
+
 func newSharedResourceInformer(client *dynamicclientset.ResourceClient, defaultResyncPeriod time.Duration, close func()) *sharedResourceInformer {
 	informer := cache.NewSharedIndexInformer(
 		&cache.ListWatch{
-			ListFunc:  client.List,
+			ListFunc:  castToCacheListFunc(client.List),
```

Comment:
Since we only need to do this once, I don't think it warrants all the stuff broken out above. It should be possible to write this concisely as something like this:

```go
ListFunc: func(opts metav1.ListOptions) (runtime.Object, error) {
  return client.List(opts)
}
```

Note that you don't need to cast the return value because a type that implements an interface is already assignable to a variable of that interface type.

Reply:
This is a much better way to do it! Thanks for specifying that the casting is not necessary.